src/isomorphic/classic/types/ReactPropTypes.js

```diff
@@ -144,6 +144,12 @@ function createChainableTypeChecker(validate) {
     if (props[propName] == null) {
       var locationName = ReactPropTypeLocationNames[location];
       if (isRequired) {
+        if (props[propName] === null) {
+          return new Error(
+            `Required ${locationName} \`${propFullName}\` was specified in ` +
```

Comment:
Can we do

```
The ${locationName} `${propFullName}` is required in ${componentName}, but its value is null.
```

and replace the other message with "…its value is undefined."?

lgtm. Can you send me a diff internally to `html/shared/core/createWarning.js` adding the new wording so we don't forget?